specs/DesignDiscussions/InputValidation.md

```diff
@@ -0,0 +1,260 @@
+
+- We do validate on focus lost, so a user who has navigated to and away from an input will have seen the validation error, even if they subsequently hid it using a toggle or show card action. This means that in scenarios where the input is not required but has a regex or other validation (such as the personalization example above) the user will already be aware that their field has not validated correctly. The scenario where a **required** input is not shown may be confusing to a user, but one could make the case that it's questionable card design to hide required fields behind toggle or show card actions.
+
+- Allowing the card author to explicitly specify inputs allows them to override this default behavior to configure the validation as appropriate for their scenario.
+
+### Validation and Actions
+
+#### Action Types
+Although submit actions are the most obvious case for input validation, we will support validation on all action types. The `inputs` property will indicate which fields are validated.
+
+##### Submit (and Invoke) Actions
+Submit actions will default to `"inputs": "All"`. This means that all inputs on the card will be validated and sent with this submission. Card authors can adjust this default behavior by setting the `inputs` property to a different value as described above.
+
+##### Toggle and ShowCard Actions
+In many, or even most cases, it likely doesn't make sense to validate on show card or toggle actions if those actions are simply showing more details that don't need to appear on the main card. In some cases, however, either show card or toggle may be used to reveal the next step in a progression through the card experience. The wizard card above is a clear example of this. The card author may want to validate the first "page" before allowing the user to proceed to the second. Event the FoodOrder card can be viewed as an example of using show cards to progress through an interaction. If the upper portion of the card contained addition inputs, the card author may want to validate those fields before proceeding to ask additional questions about the users order. 
+
+Because by default Toggles and ShowCards don't necessarily need to validate inputs, these actions will default to `"inputs": "None"`. We will, however, support the ability to set a different value to enforce validation of some or all of the inputs before allowing the user to take one of these actions.
+
+> Question: Is it confusing to have an `inputs` property on a type that doesn't actually return those inputs (only validates them)? Would we want to add support for returning inputs on other action types? Is this an indication we need a different name?
+
+##### OpenUrl
```

Comment:
Discussion:
- This could be useful if we someday have a scenario where the URL is configured by the values of the inputs.
- If we had that, the inputs would be explicit there, so this would be redundant.
- We could have the behavior be that if there are explicit inputs they are validated in addition to anything in input.

Ultimately we're okay with having inputs on the base class.

Reply:
We went back to this issue, David thinks Inputs should only be a property on Action.Submit instead of all actions. This would make it easier to extend Inputs to all Action elements; if we consider Inputs to be necessary in all actions we can introduce it later